app/models/reservation.rb

```diff
@@ -156,11 +157,14 @@ def overlaps_with(d)
   end
 
   def flag(flag)
+    return unless FLAGS.key? flag
```

Comment:
See here I think it makes sense to just return `nil` (and same with `unflag`. Should we extract `FLAGS.key? flag` into a separate (class?) method since we use it in multiple places (e.g. `flag_exists?`)?